lesson18/main.js

```diff
@@ -192,13 +198,13 @@ const switchActiveIndicator = (e) => {
   targetIndicator.classList.add("is-active");
 };
 
+const isClickedTargetElement = (index) => index !== -1;
+
 const switchImgForIndicator = (e) => {
   const currentImg = document.querySelector(".is-show");
   const targetIndicator = e.target;
-  const indexOfTargetIndicator =
-    createArrayOfIndicators().indexOf(targetIndicator);
-    //子要素indicatorをクリックしたらその要素をアクティブにし、親要素をクリックした場合はなにもしない
-  const isClickedTargetElement = (index) => index !== -1;
+  const indexOfTargetIndicator = indicators.indexOf(targetIndicator);
+  //子要素indicatorをクリックしたらその要素をアクティブにし、親要素をクリックした場合はなにもしない
```

Comment:
Every time the parent element is clicked, the clickIndicatorEvents are executed, causing the movement on the screen to stagger.I thought it would be more performant to handle removing the parent element from the click event.

```js
const clickIndicatorEvents = (e) => {
  stopAutoPlay();
  startAutoPlay();
  switchActiveIndicator(e);
  switchImgForIndicator(e);
  switchDisableForBtn();
  getCurrentPageNum();
};
```

Reply:
Thank you for your advice :)
Here, actually I implemented this click event for only child elements using `for`, but I've changed this with Mr.Morita's advice([detail](https://github.com/yuka830/js-lessons/pull/23#discussion_r753660058))
When I got that advice I agreed with his advice, because it would be better performance to put it on the parent element rather than looping through the child element. 

But I understand your advice too. Also it needs this part now.
```
const indexOfTargetIndicator = indicators.indexOf(targetIndicator);
  //子要素indicatorをクリックしたらその要素をアクティブにし、親要素をクリックした場合はなにもしない
```

I've not decided how I implement yet but I'll think about this additional part again 🤔 When Mr.Morita gave me this advice, he didn't say it's "MUST" :) 
Thank you!

Reply:
I'd tried to find some reference code about this, but couldn't find it...
And I'd not decided what I should do, but this time I'll leave this part unchanged because I've got approve with this code.

But I've learned what happens when I add an event on the parent element in making dot pagination too.
And also thank you for reviewing!
I appreciate it :)

Reply:
IMO
Yuka-san, don't give up yet!

The fix I had in mind was something like this!
If you click on the parent element, nothing will happen, what do you think?

```js
const renderIndicator = () => {
  const pagination = createElementWithClassName("div", "indicators");
  pagination.id = "pagination";   //親要素にidを付与させる
  const fragment = document.createDocumentFragment();
  for (let i = 0; i < images.length; i++) {
    const indicator = createElementWithClassName("span", "indicator");
    //最初のインデックスをアクティブにする
    i === 0 && indicator.classList.add("is-active");
    fragment.appendChild(indicator);
  }
  pagination.appendChild(fragment);
  slideshowWrap.insertAdjacentElement("afterend", pagination);
};
```
and

```js
const clickIndicatorEvents = (e) => {
  const parent = document.getElementById("pagination");
//クリックしたのが親要素だったら何もしない
  if (e.target === parent) {
    return;
  }
  stopAutoPlay();
  startAutoPlay();
  switchActiveIndicator(e);
  switchImgForIndicator(e);
  switchDisableForBtn();
  getCurrentPageNum();
};
```
[codesandbox](https://codesandbox.io/s/lesson18-create-automatic-slideshow-forked-nljvj?file=/main.js:5344-5626)

Reply:
I thought you meant the way to add event but I totally misunderstood, I'm sorry about that.
And thank you so much for giving nice advice. I've changed the code with your advice and deleted some code I implemented for avoiding parent element when I click indicators before.
Now it looks more simply and easier code🚀 🚀
Much appreciated!
[abf93ac](https://github.com/yuka830/js-lessons/pull/24/commits/abf93ac5819844a2284cf6df3aa6457a5225cc6f)